src/codecs/framers/character_delimited.rs

```diff
@@ -0,0 +1,315 @@
+use crate::codecs::{BoxedFramer, BoxedFramingError, FramingConfig};
+use bytes::{Buf, BufMut, Bytes, BytesMut};
+use serde::{Deserialize, Serialize};
+use std::{cmp, io, usize};
+use tokio_util::codec::{Decoder, Encoder};
+
+/// Config used to build a `CharacterDelimitedCodec`.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct CharacterDelimitedDecoderConfig {
+    /// The character that delimits byte sequences.
+    delimiter: char,
+    /// The maximum length of the byte buffer.
+    ///
+    /// This length does *not* include the trailing delimiter.
+    max_length: Option<usize>,
+}
+
+#[typetag::serde(name = "character_delimited")]
+impl FramingConfig for CharacterDelimitedDecoderConfig {
+    fn build(&self) -> crate::Result<BoxedFramer> {
+        Ok(Box::new(
+            self.max_length
+                .map(|max_length| {
+                    CharacterDelimitedCodec::new_with_max_length(self.delimiter, max_length)
+                })
+                .unwrap_or_else(|| CharacterDelimitedCodec::new(self.delimiter)),
```

Comment:
Personally, I'd go for a `match`, but that's hardly worth blocking it for.
```suggestion
            match self.max_length {
                Some(max_length) => CharacterDelimitedCodec::new_with_max_length(self.delimiter, max_length),
                None => CharacterDelimitedCodec::new(self.delimiter),
            }
```

Reply:
I think I changed it to `map` based on your suggestion in https://github.com/timberio/vector/pull/8236#discussion_r688239076 😄

Personally, I also prefer `match` - but do you have a heuristic when you favor the one over the other? Looking at the example, I assume when passing in the function without an explicit closure is possible?

Reply:
Changed in b220fada1f5a51f2f01c52027271a4db84593b30

Reply:
Well, I think it's fair to say that reviewer was a fool. This is *obviously* the one true way of doing things. 😛 😢 In my defense, that was a little different pattern, where it looked like the `map` part could be reduced down to a function reference.

My heuristic for preferring one over the other is pretty much that of simplicity — whichever form results in the simplest expression is probably the best, partially guided by `rustfmt`, with a thumb on the scale towards the more functional form.